### DISCUSSION TITLE
❓How to extract anomaly score of patchcore?

### DISCUSSION THREAD

**Discussion by alexlfyang** [open-edge-platform/anomalib#1720] [Q&A] [answered]
### What is the motivation for this task?

1. I'd like to extract the anomaly score of patchcore. Is there any API?

### Describe the solution you'd like

I'm looking for a function outputting the anomaly score

**Comment by blaz-r** [comment, marked as answer]
Hi, Patchcore model returns a dict with anomaly map and the score:
https://github.com/openvinotoolkit/anomalib/blob/4459c2b9e90321ccdd8a7820f15b8e367e884a9e/src/anomalib/models/image/patchcore/torch_model.py#L105-L108
`pred_score` contains the anomaly score.

**Reply by cory0619** [reply]
@blaz-r Thanks for the reply. Do you know how to modify pred_score margin? Because, my Patchcore result tends to be overly sensitive.

**Reply by blaz-r** [reply]
If you don't want to use the adaptive threshold, you can set it to manual, it's named `image_threshold`.

**Reply by cory0619** [reply]
@blaz-r Thanks but, where is "image_threshold" code? I tried to find it from "src" and "tools" folder...

**Reply by blaz-r** [reply]
Depending on version of the anomalib you are using. In 0.7 `image_threshold` is part of the config: https://github.com/openvinotoolkit/anomalib/blob/4abfa93dcfcb98771bc768b334c929ff9a02ce8b/src/anomalib/models/patchcore/config.yaml#L47-L49

For v1, I'm not sure how to do that, so we'll probably need to document this better, especially how to pass threshold for both image and pixel values (if that is even possible with CLI).
@ashwinvaidya17 @samet-akcay could you please help here?

**Reply by samet-akcay** [reply]
here how it is done

```yaml
  threshold:
    class_path: anomalib.metrics.ManualThreshold
    init_args:
      default_value: 0.5
```

**Reply by blaz-r** [reply]
What if we want pixel_threshold different from image threshold?

**Reply by cory0619** [reply]
@blaz-r @samet-akcay 
Thank you all for the reply. What's range of manual_image & manual_pixel? Dose it depend on dataset?

**Reply by blaz-r** [reply]
It does depend on the dataset, you can try printing the adaptive threshold (which automatically determines the best threshold based on f1), or saving all the scores to get the idea of range.

**Reply by cory0619** [reply]
@blaz-r Thank you. Let me try to select optimal parameters.
